### PR TITLE
Structured logging for docker image manager

### PR DESCRIPTION
**Pull request by sparrc**
### Summary

Make the docker image manager logs more readable and traceable by using structured logging and adding fields with consistent metadata about the image in each log line.

Also moved a few redundant log messages from INFO to DEBUG 

### Description for the changelog

NA

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/engine/docker_image_manager.go [hunk 1]
```diff
@@ -272,7 +273,7 @@ func (imageManager *dockerImageManager) removeImageState(imageStateToBeRemoved *
 	for i, imageState := range imageManager.imageStates {
 		if imageState.Image.ImageID == imageStateToBeRemoved.Image.ImageID {
 			// Image State found; hence remove it
-			seelog.Infof("Removing Image State: [%s] from Image Manager", imageState.String())
+			logger.Debug("Image Manager: removing image state", imageState.Fields())
```
Comment (sparrc) [inline agent/engine/docker_image_manager.go:276]: this is not removing the image state from the instance, only from the agent's internal tracking object. This seems confusing to have as a public INFO message so making it debug

File: agent/engine/docker_image_manager.go [hunk 2]
```diff
@@ -282,14 +283,14 @@ func (imageManager *dockerImageManager) removeImageState(imageStateToBeRemoved *
...
-		seelog.Debugf("Image Manager: Empty state!")
+		logger.Debug("Image Manager: Empty state!")
 		// no image states present in image manager
 		return nil
 	}
 	var imagesForDeletion []*image.ImageState
 	for _, imageState := range imageManager.imageStatesConsideredForDeletion {
 		if imageManager.isImageOldEnough(imageState) && imageState.HasNoAssociatedContainers() {
-			seelog.Infof("Candidate image for deletion: [%s]", imageState.String())
+			logger.Debug("Candidate image for deletion", imageState.Fields())
```
Comment (sparrc) [inline agent/engine/docker_image_manager.go:293]: these messages are logged multiple times when the image is actually removed, and then again on each "removal" loop, so making it debug

**Review by danehlim — APPROVED** [review]

File: agent/api/container/container.go [hunk 3]
```diff
@@ -26,6 +26,7 @@ import (
 	apierrors "github.com/aws/amazon-ecs-agent/agent/api/errors"
 	resourcestatus "github.com/aws/amazon-ecs-agent/agent/taskresource/status"
 	"github.com/aws/amazon-ecs-agent/ecs-agent/credentials"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger"
```
Comment (chienhanlin) [inline agent/api/container/container.go:29]: Q. We have some log field defined in [logger/field/constants.go](https://github.com/aws/amazon-ecs-agent/blob/dev/ecs-agent/logger/field/constants.go), should be add new useful fields there and reuse them? Thanks!

File: agent/engine/image/types.go [hunk 4]
```diff
@@ -188,5 +188,27 @@ func (imageState *ImageState) String() string {
...
+	imageID := ""
+	names := []string{}
+	size := int64(-1)
+	if imageState.Image != nil {
+		imageID = imageState.Image.ImageID
+		names = imageState.Image.Names
+		size = imageState.Image.Size
+	}
+
+	return logger.Fields{
```
[the remark this answers is not in the document]
Comment (sparrc) [reply agent/engine/image/types.go:205]: good call, done!

**Review by chienhanlin — APPROVED** [review]
Thanks!

File: agent/engine/docker_image_manager.go [hunk 5]
```diff
@@ -648,24 +662,28 @@ func (imageManager *dockerImageManager) removeImage(ctx context.Context, leastRe
 
 func (imageManager *dockerImageManager) deleteImage(ctx context.Context, imageID string, imageState *image.ImageState) {
 	if imageID == "" {
-		seelog.Errorf("Image ID to be deleted is null")
+		var fields logger.Fields
+		if imageState != nil {
+			fields = imageState.Fields()
+		}
+		logger.Error("Image ID to be deleted is null", fields)
```
Comment (singholt) [inline agent/engine/docker_image_manager.go:669]: logger.Error here should be in an else block?

File: agent/engine/docker_image_manager.go [hunk 6]
```diff
@@ -480,24 +487,29 @@ func (imageManager *dockerImageManager) removeNonECSImages(ctx context.Context,
...
+			"repoTags":           image.RepoTags,
+		}
 		if len(image.RepoTags) > 1 {
-			seelog.Debugf("Non-ECS image has more than one tag Image: %s (Tags: %s)", image.ImageID, image.RepoTags)
+			logger.Debug("Non-ECS image has more than one tag", fields)
 			for _, tag := range image.RepoTags {
 				err := imageManager.client.RemoveImage(ctx, tag, dockerclient.RemoveImageTimeout)
 				if err != nil {
-					seelog.Errorf("Error removing RepoTag (ImageID: %s, Tag: %s) %v", image.ImageID, tag, err)
+					logger.Error("Error removing non-ECS RepoTag", fields, logger.Fields{field.Error: err})
```
[the remark this answers is not in the document]
Comment (sparrc) [reply agent/engine/docker_image_manager.go:500]: yes, will add that

**Pull request merged**